keras_nlp/layers/start_end_packer.py

```diff
@@ -0,0 +1,158 @@
+        sequence_length: int. The desired output length.
+        start_value: int/str. The ID or token that is to be placed at the start
+            of each sequence. The dtype must match the dtype of the input
+            tensors to the layer. If None, no start value will be added.
+        end_value: int/str. The ID or token that is to be placed at the end of
+            each input segment. The dtype must match the dtype of the input
+            tensors to the layer. If None, no end value will be added.
+        pad_value: int/str. The ID or token that is to be placed into the
+            unused positions after the last segment in the sequence. If None,
+            0 or "" will be added depending on the dtype of the input tensor.
+
+    Examples:
+
+    Unbatched input (int).
+    >>> input_data = tf.constant([5, 6, 7])
+    >>> start_end_packer = keras_nlp.layers.StartEndPacker(
+    ...     sequence_length=7, start_value=1, end_value=2, pad_value=0
+    ... )
+    >>> start_end_packer(input_data)
+    <tf.Tensor: shape=(7,), dtype=int32, numpy=
+    array([1, 5, 6, 7, 2, 0, 0], dtype=int32)>
+
+    Batched input (int).
+    >>> input_data = tf.ragged.constant([[5, 6, 7], [8, 9, 10, 11]])
+    >>> start_end_packer = keras_nlp.layers.StartEndPacker(
+    ...     sequence_length=6, start_value=1, end_value=2, pad_value=0
+    ... )
+    >>> start_end_packer(input_data)
+    <tf.Tensor: shape=(2, 6), dtype=int32, numpy=
+    array([[ 1,  5,  6,  7,  2,  0],
+           [ 1,  8,  9, 10, 11,  2]], dtype=int32)>
+
+    Unbatched input (str).
+    >>> input_data = tf.constant(["this", "is", "fun"])
+    >>> start_end_packer = keras_nlp.layers.StartEndPacker(
+    ...     sequence_length=6, start_value="<s>", end_value="</s>",
+    ...     pad_value="<pad>"
+    ... )
+    >>> start_end_packer(input_data)
+    <tf.Tensor: shape=(6,), dtype=string, numpy=
+    array([b'<s>', b'this', b'is', b'fun', b'</s>', b'<pad>'], dtype=object)>
+
+    Batched input (str).
+    >>> input_data = tf.ragged.constant([["this", "is", "fun"], ["awesome"]])
+    >>> start_end_packer = keras_nlp.layers.StartEndPacker(
+    ...     sequence_length=6, start_value="<s>", end_value="</s>",
+    ...     pad_value="<pad>"
+    ... )
+    >>> start_end_packer(input_data)
+    <tf.Tensor: shape=(2, 6), dtype=string, numpy=
+    array([[b'<s>', b'this', b'is', b'fun', b'</s>', b'<pad>'],
+           [b'<s>', b'awesome', b'</s>', b'<pad>', b'<pad>', b'<pad>']],
+          dtype=object)>
+    """
+
+    def __init__(
+        self,
+        sequence_length,
+        start_value=None,
+        end_value=None,
+        pad_value=None,
+        name=None,
+        **kwargs,
+    ):
+        super().__init__(name=name, **kwargs)
+
+        self.sequence_length = sequence_length
+        self.start_value = start_value
+        self.end_value = end_value
+        self.pad_value = pad_value
+
+    def call(self, inputs):
+        input_is_tensor = isinstance(inputs, tf.Tensor)
```

Comment:
Before this should probably do the same

```
if not isinstance(inputs, tf.Tensor) or isinstance(inputs, tf.RaggedTensor):
    inputs = tf.conver_to_tensor(inputs)
```

That will allow us to support things like numpy or list inputs. Useful for demos.

Reply:
👍🏼 . I've removed the `else` block below:

```
        else:
            raise ValueError(
                "Input must be of type `tf.Tensor` or `tf.RaggedTensor`, "
                f"but got {type(inputs)}"
            )
```

Reply:
One minor thing - I would rename this to `input_is_dense` to match `input_is_ragged`.

Reply:
Done!